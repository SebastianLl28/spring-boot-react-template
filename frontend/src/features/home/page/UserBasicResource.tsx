interface Props {
  name: string
  id: number
}

const UserBasicResource = ({ name, id }: Props) => {
  return (
    <div>
      <h3>Hola {name}</h3>
      <p>id: {id}</p>
    </div>
  )
}

export default UserBasicResource

