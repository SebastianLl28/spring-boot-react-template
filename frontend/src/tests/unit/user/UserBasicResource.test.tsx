import UserBasicResource from '@/features/home/page/UserBasicResource'
import { userResponse } from '@/tests/fixture/user.fixture'
import { render } from '@/tests/utils/render'
import { screen } from '@testing-library/react'
import { test, expect, beforeEach, describe } from 'vitest'

describe('UserBasicResource', () => {
  let container: HTMLElement
  const { id, username } = userResponse

  beforeEach(() => {
    const renderResult = render(<UserBasicResource name={username} id={id} />)
    container = renderResult.container
  })

  test('should render correctly', () => {
    expect(screen).toBeTruthy()
  })

  test('should render name', () => {
    const text = screen.getByText(`Hola ${username}`)
    expect(text).toBeTruthy()
  })

  test('should render id', () => {
    const text = screen.getByText(`id: ${id}`)
    expect(text).toBeTruthy()
  })

  test('should match snapshot', () => {
    expect(container).toMatchSnapshot()
  })
})

