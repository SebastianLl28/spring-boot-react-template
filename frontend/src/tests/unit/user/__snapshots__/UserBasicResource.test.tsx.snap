// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserBasicResource > should match snapshot 1`] = `
<div>
  <div>
    <h3>
      Hola 
      test
    </h3>
    <p>
      id: 
      1
    </p>
  </div>
</div>
`;
